app/controllers/api/feedback_submissions_controller.rb

```diff
@@ -0,0 +1,21 @@
+module Api
+  class FeedbackSubmissionsController < ApiController
+    def create
+      @feedback = FeedbackSubmission.new(feedback_params)
+
+      if @feedback.save
+        ZendeskTicketsJob.perform_later(@feedback)
+      else
+        render status: 422, json: { messages: @feedback.errors.full_messages }
```

Comment:
I realised afterwards that this is probably what you meant. I still think that is better to have one source for the validations where possible, that way we don't have to update more than 1 codebase when changing validations. Also 500 in my view are errors in the server side, a client shouldn't be able to trigger them. Lets go over some of the code I have when you are in.

Regarding json-api, my main worry is APIs consistency long term. I agree with you that initially might seem heavier than needed, however I think it can be beneficial long term:
- Have consistent API's across the the different micro services, which will make it easier to  build new services, understand the API of other services, or making changes to them.
- As this API evolves (or others), it might require design breaking changes. From experience it is more costly to introduce breaking changes later on. By using a standard we can piggy back on design decisions that have had a lot of thought on them.
- Code reuse as there is good open source support across different languages and frameworks.

Let's discuss when you are in.